cellengine/resources/gate.py

```diff
@@ -193,81 +152,345 @@ def create(
+                been applied).
+            label (float, optional): Position of the label. Defaults to the
+                midpoint of the gate. gid (str): Group ID of the gate, used for
+                tailoring. If this is not specified, then a new Group ID will
+                be created. To create a tailored gate, the gid of the global
+                tailored gate must be specified.
+            locked (bool, optional): Prevents modification of the gate via the
+                web interface. tailored_per_file (bool): Whether or not this
+                gate is tailored per FCS file.
+            fcs_file_id (str, optional): ID of FCS file, if tailored per file.
+                Use ``None`` for the global gate in a tailored gate group. If
+                specified, do not specify ``fcs_file``.
+            fcs_file (str, optional): Name of FCS file, if tailored per file.
+                An attempt will be made to find the file by name. If zero or
+                more than one file exists with the name, an error will be
+                thrown. Looking up files by name is slower than using the ID,
+                as this requires additional requests to the server. If
+                specified, do not specify ``fcs_file_id``.
 
-class PolygonGate(Gate):
-    """Basic concrete class for polygon gates"""
+        Returns:
+            A RectangleGate object.
+
+        Example:
+            ```python
+            experiment.create_rectangle_gate(x_channel="FSC-A", y_channel="FSC-W",
+            name="my gate", 12.502, 95.102, 1020, 32021.2)
+            cellengine.Gate.create_rectangle_gate(experiment_id, x_channel="FSC-A",
+            y_channel="FSC-W", name="my gate", x1=12.502, x2=95.102, y1=1020,
+            y2=32021.2, gid=global_gate.gid)
+            ```
+        """
+        kwargs = get_args_as_kwargs(cls, locals())
+        params = {
+            k: kwargs.pop(k) for k in ["create_population", "parent_population_id"]
+        }
+
+        gate = cls._format(**kwargs)
+        return ce.APIClient().create(Gate(id=None, **gate), **params)  # type: ignore
 
     @classmethod
-    @doc_inherit(format_polygon_gate)
+    @exception_handler
+    def _format(cls, **kwargs):
+        """Get relevent kwargs and shape into a gate model"""
+
+        kwargs = remove_keys_with_none_values(kwargs)
+
+        x1 = kwargs.get("x1") or kwargs.get("model", {}).get("rectangle", {}).get("x1")
+        x2 = kwargs.get("x2") or kwargs.get("model", {}).get("rectangle", {}).get("x2")
+        y1 = kwargs.get("y1") or kwargs.get("model", {}).get("rectangle", {}).get("y1")
+        y2 = kwargs.get("y2") or kwargs.get("model", {}).get("rectangle", {}).get("y2")
+        label = kwargs.get("label") or kwargs.get("model", {}).get(
+            "label",
+            [
+                mean([x1, x2]),
+                mean([y1, y2]),
+            ],
+        )
+
+        model = {
+            "locked": kwargs.get("model", {}).get("locked", False),
```

Comment:
I don't see this passed via `model` either. just `kwargs.get("locked", False)`?

Reply:
How do you mean? `model` is used on l. 250; this syntax returns `kwargs['model']['locked']` if it exists and otherwise `False`. `kwargs['locked']` isn't a valid option because the returned JSON has 'locked' nested inside 'model'

Reply:
IDK what I was looking at, this is correct. :)

Reply:
Actually I see what I was looking at. `def create()` on line 137R has a named `locked` argument but no `model` argument, so I think this is always effectively `{}.get("locked", False)`. Make sense?

Reply:
Yup, I see it, thanks! fixed